pkg/server/plugin/upstreamauthority/awssecret/awssecret_test.go

```diff
@@ -3,217 +3,358 @@ package awssecret
-func (as *Suite) TestAWSSecret_GetPluginInfo() {
-	res, err := as.plugin.GetPluginInfo(ctx, &spi.GetPluginInfoRequest{})
-	as.Require().NoError(err)
-	as.Require().NotNil(res)
-}
-
-func (as *Suite) newAWSUpstreamAuthority() upstreamauthorityv0.UpstreamAuthorityClient {
-	config := Config{
-		KeyFileARN:      "key",
+	successConfiguration := Configuration{
+		Region:          "region_1",
 		CertFileARN:     "cert",
-		AccessKeyID:     "keyid",
-		SecretAccessKey: "accesskey",
+		KeyFileARN:      "key",
+		AccessKeyID:     "access_key_id",
+		SecretAccessKey: "secret_access_key",
+		SecurityToken:   "security_token",
+		AssumeRoleARN:   "assume_role_arn",
 	}
 
-	var err error
-	as.client, err = newFakeSecretsManagerClient(nil, "region")
-	as.Require().NoError(err)
-
-	m := newPlugin(newFakeSecretsManagerClient)
-	m.hooks.clock = as.clock
+	for _, tt := range []struct {
+		test                    string
+		makeConfigFail          bool
+		configuration           Configuration
+		csr                     []byte
+		preferredTTL            time.Duration
+		expectCode              codes.Code
+		expectMsgPrefix         string
+		expectX509CASpiffeID    string
+		expectedX509Authorities []*x509.Certificate
+		expectTTL               time.Duration
+	}{
+		{
+			test:                    "valid CSR",
+			configuration:           successConfiguration,
+			csr:                     makeCSR("spiffe://example.org"),
+			preferredTTL:            x509svid.DefaultUpstreamCATTL + time.Hour,
+			expectTTL:               x509svid.DefaultUpstreamCATTL + time.Hour,
+			expectX509CASpiffeID:    "spiffe://example.org",
+			expectedX509Authorities: x509Authority,
+		},
+		{
+			test:                    "using default ttl",
+			configuration:           successConfiguration,
+			csr:                     makeCSR("spiffe://example.org"),
+			expectTTL:               x509svid.DefaultUpstreamCATTL,
+			expectX509CASpiffeID:    "spiffe://example.org",
+			expectedX509Authorities: x509Authority,
+		},
+		{
+			test:            "configuration fail",
+			makeConfigFail:  true,
+			configuration:   successConfiguration,
+			csr:             makeCSR("spiffe://example.org"),
+			expectCode:      codes.FailedPrecondition,
+			expectMsgPrefix: "upstreamauthority(awssecret): not configured",
+		},
+		{
+			test:            "unable to sing CSR",
+			configuration:   successConfiguration,
+			csr:             []byte{1},
+			expectCode:      codes.Internal,
+			expectMsgPrefix: "upstreamauthority(awssecret): unable to sign CSR: unable to parse CSR",
+		},
+	} {
+		tt := tt
+		t.Run(tt.test, func(t *testing.T) {
+			p := new(Plugin)
+			p.hooks.clock = clk
+			p.hooks.newClient = newFakeSecretsManagerClient
+
+			ua := new(upstreamauthority.V1)
+			plugintest.Load(t, builtin(p), ua,
+				plugintest.ConfigureJSON(tt.configuration),
+				plugintest.CoreConfig(catalog.CoreConfig{
+					TrustDomain: spiffeid.RequireTrustDomainFromString("example.org"),
+				}),
+			)
+
+			if tt.makeConfigFail {
```

Comment:
maybe instead we can either have a `skipConfigure` flag or don't pass `ConfigureJSON` option if tt.configuration is empty? reaching into the plugin to set this to nil seems weird.